packages/odyssey-react-mui/src/labs/TopNav.tsx

```diff
@@ -0,0 +1,459 @@
+              >
+                {label}
+              </TopNavItemLabelContainer>
+            </Link>
+          )
+        }
+      </TopNavListItemContainer>
+    );
+  }, [
+    id,
+    label,
+    href,
+    target,
+    onClick,
+    isDisabled,
+    topNavItemContentKeyHandler,
+    odysseyDesignTokens,
+  ]);
+
+  return TopNavItemContent;
+};
+
+const LinkAndProfileWrapper = styled.div({
+  display: "flex",
+  alignItems: "center",
+  marginLeft: "auto",
+});
+
+const AdditionalLinkContainerWithBorder = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  display: "flex",
+  alignItems: "center",
+  marginRight: odysseyDesignTokens.Spacing3,
+  borderRight: `${odysseyDesignTokens.BorderWidthMain} solid ${odysseyDesignTokens.HueNeutral200}`,
+}));
+
+const LinkContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  paddingRight: odysseyDesignTokens.Spacing3,
+  "& a": {
+    color: `${odysseyDesignTokens.TypographyColorHeading} !important`,
+  },
+}));
+
+const TopNavContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  display: "flex",
+  alignItems: "center",
+  backgroundColor: odysseyDesignTokens.HueNeutralWhite,
+  height: odysseyDesignTokens.Spacing9,
+}));
+
+const LogoContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  display: "flex",
+  alignItems: "center",
+  padding: `0 ${odysseyDesignTokens.Spacing9} 0 ${odysseyDesignTokens.Spacing5}`,
+}));
+
+const SearchFieldContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  width: "350px",
+  padding: `${odysseyDesignTokens.Spacing2} ${odysseyDesignTokens.Spacing3}`,
+}));
+
+const AdditionalNavItemContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  padding: `0 ${odysseyDesignTokens.Spacing3}`,
+}));
+
+const TopNav = ({
+  hasLogo,
+  SearchFieldComponent,
+  topNavLinkItems,
+  AdditionalNavItemComponent,
+  settingsLink,
+  helpLink,
+  userProfile,
+}: TopNavProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const LogoStyles = useMemo(
+    () => ({
+      fontSize: odysseyDesignTokens.TypographyScale6,
+    }),
+    [odysseyDesignTokens],
+  );
+
+  const LogoWordmarkStyles = useMemo(
+    () => ({
+      width: "55px",
+      height: "20px",
+      paddingLeft: odysseyDesignTokens.Spacing2,
+    }),
+    [odysseyDesignTokens],
+  );
```

Comment:
We should see if we could get these values lined up with design tokens somehow.

Reply:
Couldn't find any tokens that might work for these props

Reply:
@ganeshsomasundaram-okta We should use `Spacing9` for the width. And we probably don't need to set a height here

Reply:
Also curious why it needs padding here. I'd think the containing element would provide any necessary padding

Reply:
thanks @bryancunningham-okta I'll use the Spacing9, it is slightly smaller than the figma spec but it should be close enough. Also, there is a padding/gap between the aura and the workmark in the design. I don't have a container around the aura/workmark.
![image](https://github.com/user-attachments/assets/b0d104b9-ec05-4d97-966a-2874fff12811)

Reply:
Ah, got ya. Can we use margin instead then? Makes it more clear what the intention is. And use the logical css property
```css
marginInlineStart: odysseyDesignTokens.Spacing2
```

Reply:
removed the logo for now to unlock this PR. We are waiting from the design team to confirm where the logo would go (between top-nav and side-nav). There are some inconsistencies on the logo behavior/location between enduser & admin apps that design team needs to consolidate.